src/assets/scripts/client/trafficGenerator/buildPreSpawnAircraft.js

```diff
@@ -92,25 +92,31 @@ const _calculateDistancesAlongRoute = (waypointModelList, airport) => {
     let distanceFromClosestFixToAirspaceBoundary = 0;
     let totalDistance = 0;
 
-    for (let i = 0; i < waypointModelList.length; i++) {
+    // Iteration started at index 1 to ensure two elements are available. It is
+    // already an expectation that aircraft must have two waypoints, so this
+    // should not be a problem here.
+    for (let i = 1; i < waypointModelList.length; i++) {
         const waypoint = waypointModelList[i];
-        const waypointPosition = waypoint.relativePosition;
-        let previousWaypoint = waypoint;
-        let previousPosition = waypoint.relativePosition;
+        const previousWaypoint = waypointModelList[i - 1];
 
-        if (i > 0) {
-            previousWaypoint = waypointModelList[i - 1];
-            previousPosition = previousWaypoint.relativePosition;
+        // TODO: Refactor to avoid continue statement
+        if (waypoint.isVector || previousWaypoint.isVector) {
+            continue;
         }
 
-        if (isWithinAirspace(airport, waypointPosition) && i > 0) {
+        const waypointPosition = waypoint.relativePosition;
+        const previousPosition = waypoint.relativePosition;
+
+        if (isWithinAirspace(airport, waypointPosition)) {
             distanceFromClosestFixToAirspaceBoundary = nm(calculateDistanceToBoundary(airport, previousPosition));
+            totalDistance += distanceFromClosestFixToAirspaceBoundary;
 
-            continue;
+            break;
```

Comment:
I did. No idea why we would continue. We don't want any of the distances _within_ the airspace border, since we don't spawn there. So we should break, but we _do_ want to add this value to the total, so I added an adder within the breaker block. (<-- now that's a cool sounding sentence)